SpatialGDKEditorToolbar/Source/Private/SpatialGDKEditorGenerateSnapshot.cpp

```diff
@@ -118,12 +119,24 @@ bool CreateGlobalStateManager(Worker_SnapshotOutputStream* OutputStream)
 	ComponentWriteAcl.Add(SpatialConstants::UNREAL_METADATA_COMPONENT_ID, UnrealServerPermission);
 	ComponentWriteAcl.Add(SpatialConstants::ENTITY_ACL_COMPONENT_ID, UnrealServerPermission);
 	ComponentWriteAcl.Add(SpatialConstants::GLOBAL_STATE_MANAGER_COMPONENT_ID, UnrealServerPermission);
+	ComponentWriteAcl.Add(SpatialConstants::GLOBAL_STATE_MANAGER_MAP_URL, UnrealServerPermission);
 
 	Components.Add(improbable::Position(Origin).CreatePositionData());
 	Components.Add(improbable::Metadata(TEXT("GlobalStateManager")).CreateMetadataData());
 	Components.Add(improbable::Persistence().CreatePersistenceData());
 	Components.Add(improbable::UnrealMetadata().CreateUnrealMetadataData());
 	Components.Add(CreateGlobalStateManagerData());
+
+	// Add the accepting players state of the GSM.
+	Worker_ComponentData MapData;
+	MapData.component_id = SpatialConstants::GLOBAL_STATE_MANAGER_MAP_URL;
+	MapData.schema_type = Schema_CreateComponentData(SpatialConstants::GLOBAL_STATE_MANAGER_MAP_URL);
+	Schema_Object* MapDataObject = Schema_GetComponentDataFields(MapData.schema_type);
+	Schema_Object* MapURLObject = Schema_AddObject(MapDataObject, 1);
+	AddStringToSchema(MapURLObject, 1, TEXT("default"));
```

Comment:
Do we check against this `default` value anywhere?

Reply:
Added a `// TODO: Fill this with the map name of the map the snapshot is being generated for.` Although it shouldn't make a huge difference. The server which connects will write it's current map to that component and so for clients it will be correct.